### DISCUSSION TITLE
Nullable Boolean and DateTimeBoolean field

### DISCUSSION THREAD

**Discussion by ziming** [laravel/nova-issues#4537] [Ideas & Feature Requests]
I think these 2 types of boolean field will be very helpful

Nullable Boolean Field
It will be useful if we can set a boolean field value to true, false or null (unknown), as for how it looks like in the index and detail view, I guess it can be a Grey Circle with no X or Tick inside. like this O

Datetime Boolean Field

Back in Nova 2 or 1 there is this very popular Datetime Boolean 3rd party package that is no longer maintained
https://github.com/antonioribeiro/nova-boolean-datetime-field

In index view, it's like boolean field, in edit view it appears like datetime field then in detail view it shows the boolean circle and the datetime beside it.

**Comment by davidhemphill** [comment]
Boolean fields are designed to display the content from a boolean column type. `null` is not a valid boolean type, being that it's neither `true` nor `false`. 

There are several ways to get the same functionality from that package inside of Nova yourself. If you can let us know a little more about what you're trying to do we can suggest options to help.

**Comment by luciantugui** [comment]
but boolean field which is populated with null from the db (nullable column), evaluates to false, and display a red circle. making this boolean field filterable, behaves inconsistent

**Reply by davidhemphill** [reply]
In most databases, `null` equates to `false`. For example, in MySQL, 0 or NULL means false and anything else means true. The default truth value from a boolean operation is 1. It's recommended to consistently set the value to the desired one when inserting the data. Otherwise, we're going to lean on the defaults of the underlying database driver here for the value. You can always use the `resolveUsing` field helper to choose different logic for that use case.

**Reply by luciantugui** [reply]
but a boolean filter works incorrectly with null values from the DB! that is a bug. I removed the nullable column, but that is not always doable

**Reply by leovalen** [reply]
> In MySQL, 0 or NULL means false and anything else means true.

This applies to comparisons, not the values themselves. NULL is absolutely not the same as 0 or false in MySQL.

A nullable boolean field should be displayed with some kind of indicator that the value is missing when null, e. g. a dash, like other nullable fields in Nova.
